android/app/src/main/java/au/com/robin/sms/MainActivity.kt

```diff
@@ -0,0 +1,69 @@
+package au.com.robin.sms
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import au.com.robin.sms.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var smsInterfaceBinding: ActivityMainBinding
+    private val PERMISSION_REQUEST_CODE = 1
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        smsInterfaceBinding = ActivityMainBinding.inflate(layoutInflater)
+        val view = smsInterfaceBinding.root
+        setContentView(view)
+
+        smsInterfaceBinding.btSend.setOnClickListener {
+            val phoneNo = smsInterfaceBinding.etPhoneNo.text?.toString()
+            val message = smsInterfaceBinding.etMessage.text?.toString()
+
+            // Check for permission
+            if (checkPermission()) {
+                // create an intent to let client choose which sim card slot
+                // for now, we use sim card 2. Normally, user will be shown options to choose from.
+                val intent = Intent()
+                intent.putExtra("slot", 1)
+                Log.d("onStart, button", "Sending SMS")
+                val smsManager = SmsSend.getSmsManager(this, intent)
+                smsManager?.sendTextMessage(phoneNo, null, message, null, null)
+            } else {
+                requestPermission()
```

Comment:
I would like all the permission requests and checks done on app launch, so that when you press the 'send' button, message must assume to work. So you just write:

```kt
smsManager.sendTextMessage(phoneno, ...) # without the question mark
```

Reply:
Okay, got it. I'll try to make that work